confmap/expand.go

```diff
@@ -43,6 +43,20 @@ func (mr *Resolver) expandValueRecursively(ctx context.Context, value any) (any,
 
 func (mr *Resolver) expandValue(ctx context.Context, value any) (any, bool, error) {
 	switch v := value.(type) {
+	case expandedValue:
+		expanded, changed, err := mr.expandValue(ctx, v.Value)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if _, ok := expanded.(expandedValue); ok {
+			return expanded, changed, nil
+		}
```

Comment:
This part needs a test where we have `field: ${env:VAL1}`, `VAL1=${env:VAL2}`, `VAL2=1234`